### DISCUSSION TITLE
Change CNI post-provisioning a cluster

### DISCUSSION THREAD

**Discussion by nitays** [k3s-io/k3s#8101] [Q&A]
**Is your feature request related to a problem? Please describe.**

Currently it's not possible to switch to different CNI )e.g. Cilium) on an up-and-running k3s cluster that already uses the default CNI (Flannel).
**Describe the solution you'd like**

An option to make k3s use a different CNI that I've installed on the cluster after I have already created and provisioned a cluster.
**Describe alternatives you've considered**

No alternatives available as far as I know
**Additional context**

**Comment by brandond** [comment]
We do not plan to add support for changing CNI on a cluster after it's been started. Like picking cluster CIDRs and other core networking choices, CNI should be decided upon before the cluster is created. While you might be able to hack at it and get it to work, it's not something we will support. Certainly it will not work without at least restarting k3s to disable flannel.

**Comment by nitays** [comment]
@brandond  I understand. I’m willing to hack on it, any chance you can at least help me with directions? Anything else that needs to happen/change in addition to restarting k3s to disable flannel?

Also, from what I understand this is considered a Top Issue on the ref page (https://www.suse.com/support/kb/doc/?id=000020904). How come a Top Issue isn’t in the roadmap?

**Comment by brandond** [comment]
> Anything else that needs to happen/change in addition to restarting k3s to disable flannel?

That's all that's necessary for K3s, although you might also want to disable the network policy controller as well. I would defer to your CNI's installation instructions for additional steps after that. 

>How come a Top Issue isn’t in the roadmap?

I don't know what "top issue" means, other than perhaps our support org wanted to highlight it? We've evaluated this in the past for all of our supported Kubernetes distributions, and determined that it is not something we want to tackle.
